frontend/src/pages/UserDashboard/UserDashboard.jsx

```diff
@@ -0,0 +1,131 @@
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {  fetchEnrolledCourses, fetchUserProfile, fetchNotEnrolledCourses } from '../../services/api';
+import styles from './UserDashboard.module.css';
+
+const UserDashboard = () => {
+  const [activeTab, setActiveTab] = useState('Home');
+  const [profileData, setProfileData] = useState(null);
+  const [coursesData, setCoursesData] = useState([]);
+  const [enrolledCoursesData, setEnrolledCoursesData] = useState([]);
+  const userId = localStorage.getItem('userId'); 
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        if (activeTab === 'Profile') {
+          if (userId) {
+            const profile = await fetchUserProfile(userId);
+            console.log(profileData);
+            setProfileData(profile);
+          }
+        } else if (activeTab === 'Home') {
+          if (userId) {  
+            const courses = await fetchNotEnrolledCourses(userId);
+            setCoursesData(courses);
+          }
+        } else if (activeTab === 'My Courses') {
+          if (userId) {            
+            const enrolledCourses = await fetchEnrolledCourses(userId);
+            setEnrolledCoursesData(enrolledCourses);
+          }
+        }
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, [activeTab, userId]);
+
+  const handleCourseClick = (course) => {
+    const isEnrolled = activeTab === 'My Courses'; 
+    navigate(`/course/${course.courseName}`, { state: { course, isEnrolled } });
+  };
+
+  const renderStars = (rating) => {
```

Comment:
nit: This should jsut be a simple react component. It doesn't need to live in a separate file, you can put it below UserDashboard.